gourmet/backends/db.py

```diff
@@ -1538,8 +1539,9 @@ def sort_groups (x,y):
             if group_order[x[0]] > group_order[y[0]]: return 1
             elif group_order[x[0]] == group_order[y[0]]: return 0
             else: return -1
-        alist=list(groups.items())
-        sorted(alist, key=sort_groups)
+
+        alist = sorted(groups.items(), key=cmp_to_key(sort_groups))
```

Comment:
This is now not a list but a generator and in the rest of the code (L1549, 1553), it's iterated over twice (second time would be empty). There's a `sort` function again with a comparison function that shouldn't work (L1550).

Reply:
Yes! `order_ings` should have a test. I don't really know what inputs are expected, so it would take me a little while to write it.

I'm game to write the test (or fix it if it's there) before merging this PR, but that will take me the day, probably.

Reply:
For me, you don't need to write a test. But I see it's broken. You can't iterate over the same generator twice.

So at least it's `alist = list(sorted(groups.items(), key=cmp_to_key(sort_groups)))`. And once L1550 is actually called, it shoud error out. But you already include `cmp_to_key`, so you can just use it.